libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1251,6 +1251,10 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 			}
 
 			var.accept(*this);
+
+			string typeName = var.annotation().type->toString(true);
+			m_errorReporter.syntaxError(var.location(),
```

Comment:
If you put this here, there will be many redundant errors above about impossible type deduction. Since we require explicit types now, type deduction reduces to only a helper for the error message.
Perhaps remove all the stuff above (as far as possible) and replace it with a simple type deduction that can also fail much more often than it previously did.

Perhaps it is better to do that in a subsequent pull request, though.